### ISSUE TITLE
💬 RFC: External Link Icons

### ISSUE THREAD

**Issue by sennyeya** [backstage/backstage#18186]
### 🔖 Need

As mentioned in #17143, links currently do not give any indication of whether they are internal or external links. Adding indicators for whether a link will open a new tab or window is good accessibility practice.

Requesting comments from:
- @backstage/maintainers 
- @backstage/discoverability-maintainers
- any other members of the community with strong opinions on this

<img width="515" alt="Screen Shot 2023-05-05 at 3 35 16 PM" src="https://user-images.githubusercontent.com/34432188/236554708-491737a0-d210-4ec9-b5bf-55365e3124e0.png">
<img width="515" alt="Screen Shot 2023-05-05 at 4 13 50 PM" src="https://user-images.githubusercontent.com/34432188/236560062-6b563867-5e53-4462-9c2b-76975f042609.png">

## New `TextLink` Component
We would create a new component that can handle this display while allowing the previous `Link` component to be used for cases outside of `Text` links, like Images and Buttons. However, aligning on a `TextLink` component requires aligning on 
1. Should icons in external links be clickable? For example, in the sidebar or in the `EntityLinksCard`?
2. What constitutes a `TextLink`? Can it contain HTML elements as children? Should it just be a raw text node? What about hover over or popovers?
3. Do we need to enforce the text-based requirement for `TextLink`? ie do all children need to be text based and if they aren't we should throw or alert in dev-mode?

## Leverage the Existing `Link` Component
The work for this is similar to that seen in #17679. Add the functionality to the existing `Link` component. This would be a fairly large change and would result in some odd behavior for non-text links. It makes more sense to be deliberate about handling text-based links separately from generic links.

## CSS based application
Similar to above and harder to debug, we should use React where possible.

## Component Drift
Changes to one component may require mirroring to the other component, for that they should share as much code as possible.

## Inconsistent Usage
Having a second component requires documentation and strong code examples and usage to encourage usage in the future. Creating a second component with very similar functionality may lead developers to be confused about where and when to use either causing confusion for users with an inconsistent interface.

### 👀 Have you spent some time to check if this RFC has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

**Comment by benjdlambert** [comment]
@backstage/discoverability-maintainers also for visibility here.

**Comment by djspotify** [comment]
My suggestion is to offer the possibility to add an icon to a link to allow for the distinction if it is needed in the context. 

When there is a group of links that are all external it might be a better design solution to add a headline to the group like "External resources" and skip the icon, as it adds visual noise to the content.

The W3C recommendation is to not open links in new tabs or windows, [linked here](https://www.w3.org/TR/WCAG20-TECHS/G200.html).

NNGroup also recommends to default to the initial functionality and check for the users context if it makes sense to change the behavior, [linked here](https://www.nngroup.com/articles/new-browser-windows-and-tabs/).

Here is an [article](https://codersblock.com/blog/external-links-new-tabs-and-accessibility/) where an accessible solution is shown.

**Comment by sennyeya** [comment]
@djspotify Apologies for the late response. I have a couple of questions from the links you shared,
1. It seemed like those are mostly for text-only links, is there any guidance for icon + text or just icon links? Those would be caught by [this](https://codersblock.com/blog/external-links-new-tabs-and-accessibility/) solution's CSS, but probably a weird UX.
2. Do we want this to be automatic? Or should users explicitly specify that they want to enable the icon in a certain context? For the recommended direction from the RFC, the assumption would be that users would use the component as normal without specifying when to add the icon (maybe with an option to remove it).

I also like the idea of adding an "External Resources" headline for the image in the RFC.

**Comment by camilaibs** [comment]
Shared with our designers team, waiting guidance 🙂

**Comment by lofredriksson** [comment]
Hej @sennyeya! First of all, apologies for the late response on our end too. @djspotify is currently on paternity leave. He is back in a couple of weeks, but I figured I'd try to offer some guidance before then.

In regards to your questions:
1. For icon + text, one option is to allow the inclusion of the external link icon to the text part. When it comes to icon only, however, I don't see a straightforward and scalable way to communicate this behavior. Thinking of the use cases for icon only external links, I wonder if the icon itself most of the time will offer enough guessability to create a mental model that clicking the icon indeed will take you to an external site?
2. The way I interpreted David's first response would be to make it optional to show the external link icon (as in default is not to show it). And I can see why. Imagine for instance a paragraph (rather than a bulleted list as shown in the example images above), I feel like it would introduce noise to the reader if every external hyperlink all of a sudden had that icon after it.

**Comment by sennyeya** [comment]
@lofredriksson Thanks for getting back to me! No worries.

1. Definitely agree, icon-only has no clear path forward. I pulled a couple of icons from the site, the first I think is clear it would redirect, but the second -- without context that this acts similarly to the first -- doesn't scream external site. For now, I think it makes sense to only tackle text-only and text + icon.
<img width="79" alt="Screen Shot 2023-09-05 at 4 19 17 PM" src="https://github.com/backstage/backstage/assets/34432188/2e71b689-0653-46bf-84c5-ef9dcc6220e6">
<img width="79" alt="Screen Shot 2023-09-05 at 4 21 59 PM" src="https://github.com/backstage/backstage/assets/34432188/eb8e1285-40e8-46fa-9fd0-13bf7a5a797c">

2. That makes sense, I'll take a stab at adding it to the linked PR.

**Comment by lofredriksson** [comment]
@sennyeya 

1. I definitely hear you! I'll make sure to loop in @djspotify on the icon only links once he is back. Perhaps him and I can come up with a better solution. But sounds good to address the first to use cases to start with.
2. Great, thank you so much!

**Comment by mark-con** [comment]
Hi,

Just wanted to add input that this is something of value that could be beneficial from feedback we have got from users.
I don't know at what stage this is in the decision to create a PR to implement it but just wanted to give a vote in favor 😄

**Comment by efenner-cambia** [comment]
Bump

**Comment by tudi2d** [comment]
We don't plan to work on this in the foreseeable future. If this blocking you somehow or you think this should be prioritized please feel free to create a new issue describing your need. Until then I will close this issue.
